### PR TITLE
Skip archiving in Glue an old table version during a table update operation

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Slack discussion: https://trinodb.slack.com/archives/CJ6UC075E/p1699927658771409

## Additional context and related issues

https://github.com/trinodb/trino/pull/14336

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by findepi** [comment]
`TestIcebergGlueCatalogSkipArchive` is now perhaps redundant

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogConfig.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 public class IcebergGlueCatalogConfig
 {
     private boolean cacheTableMetadata = true;
-    private boolean skipArchive;
+    private boolean skipArchive = true;
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogConfig.java:22]: Add some rationale explaining the `why` for the default value (unless it's obvious)

**Review by findepi — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogConfig.java (hunk 1 above)
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogConfig.java:22]: Please update the default value in metastores.md.

**Review by ebyhr — APPROVED** [review]

**Comment by findinpath** [comment]
@ebyhr / @findepi  pls run the PR with secrets.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogConfig.java (hunk 1 above)
Comment (findinpath) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogConfig.java:22]: Updated the commit comment.

**Comment by ebyhr** [comment]
/test-with-secrets sha=6fd49b892734f5b3d8c6a8f9602a033a14993e34

**Pull request merged**